### PR TITLE
Support keys on import CLI

### PR DESCRIPTION
**Pull request by travisturner**
## Overview

This is another pass at #1557. Now that #1578 has been addressed, it's safe to use this code. The commit https://github.com/pilosa/pilosa/commit/fd96d3a02ef2285aedad638bdfac50470a27a64c adds the logic to determine which node contains the writable translate file (i.e. the coordinator).

## Pull request checklist

- [x] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [x] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [ ] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [x] I have resolved any merge conflicts.
- [x] I have included tests that cover my changes.
- [x] All new and existing tests pass.

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.

### CODE REVIEW

**Comment by yuce** [comment]
Key imports for columns works, but value field imports with column keys still fails with this PR (Go and Java clients). There is no error during import, but imported data do not appear on the server.

**Comment by travisturner** [comment]
@yuce I haven't gotten to value imports (with keys) yet. I'm going to review that next. I'll start with your `bufferValuesK()` work here: https://github.com/yuce/pilosa/commits/import-value-column-keys

File: ctl/import.go [hunk 1]
```diff
@@ -108,10 +105,14 @@ func (cmd *ImportCommand) Run(ctx context.Context) error {
...
 	}
+
+	var useColumnKeys, useRowKeys bool
 	for _, index := range schema {
 		if index.Name == cmd.Index {
+			useColumnKeys = index.Options.Keys
 			for _, field := range index.Fields {
 				if field.Name == cmd.Field {
+					useRowKeys = field.Options.Keys
 					fieldType = field.Options.Type
```
Comment (yuce) [inline ctl/import.go]: A `break` after this line would be more efficient.

File: ctl/import.go [hunk 2]
```diff
@@ -108,10 +105,14 @@ func (cmd *ImportCommand) Run(ctx context.Context) error {
...
 				}
```
Comment (yuce) [inline ctl/import.go]: Similary, a `break` after this line would be more efficient to cut the loop early.

File: http/client.go [hunk 3]
```diff
@@ -291,26 +322,42 @@ func (c *InternalClient) Import(ctx context.Context, index, field string, shard
...
-		URI: *c.defaultURI,
+	// Get the coordinator node; all bits are sent to the
+	// primary translate store (i.e. coordinator).
+	nodes, err := c.Nodes(ctx)
+	if err != nil {
+		return fmt.Errorf("getting nodes: %s", err)
+	}
+	coord := getCoordinatorNode(nodes)
+	if coord == nil {
+		return fmt.Errorf("could not find coordinator node")
```
Comment (yuce) [inline http/client.go]: Would it sound better if we had "... find the coordinator..." here ?

File: ctl/import.go (hunk 1 above)
Comment (yuce) [reply ctl/import.go]: Github moved the location of the line I've made the comment. It's right after `fieldType = ...` line.

File: ctl/import.go (hunk 2 above)
Comment (yuce) [reply ctl/import.go]: Github moved the location of the line I've made the comment. It's right after the closing brace of the `for _, field ... ` loop.

**Comment by travisturner** [comment]
Fixes #1597

**Comment by yuce** [comment]
Sorry, I thought I've approved it last time.

**Comment by travisturner** [comment]
@yuce You did approve it before. If changes are added to an approved PR, then the approval is removed.
Here's the log item indicating that:
```
travisturner dismissed yuce’s stale review via 862dabe 19 hours ago
```

**Comment by travisturner** [comment]
@codysoyland I added that deprecation warning on `--string-keys`. Let me know what you think about the `InternalClient.Nodes()` method.

**Comment by travisturner** [comment]
Good point. I'll push a change for that.

**Pull request merged**